File: 02-backend/spring-boot-ecommerce/src/main/java/com/application/ecommerce/entity/Product.java
package com.application.ecommerce.entity;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name="PRODUCT")
@Data
public class Product {

	@Id
	@Column(name="id")
	@SequenceGenerator(name="sequence",sequenceName="PRODUCT_SEQ",allocationSize=1)
	@GeneratedValue(strategy=GenerationType.SEQUENCE, generator="sequence")
	private long id;


	@Column(name="sku")
	private String sku;

	@Column(name="name")
	private String name;

	@Column(name="description")
	private String description;

	@Column(name="unit_price")
	private double unitPrice;

	@Column(name="image_url")
	private String imageUrl;

	@Column(name="active")
	private boolean active;

	@Column(name="units_in_stock")
	private int unitInStock;

	@Column(name="date_created")
	@CreationTimestamp
	private Date dateCreated;

	@Column(name="last_updated")
	@UpdateTimestamp
	private Date lastUpdate;

	@ManyToOne
	@JoinColumn(name="category_id", nullable=false)
	private ProductCategory category;

	public Product()
	{

	}

	public Product(long id,String sku, String name, String description, double unitPrice, String imageUrl, boolean active, int unitInStock, Date dateCreated, Date lastUpdate) {
		this.id= id;
	    this.sku = sku;
		this.name = name;
		this.description = description;
		this.unitPrice = unitPrice;
		this.imageUrl = imageUrl;
		this.active = active;
		this.unitInStock = unitInStock;
		this.dateCreated = dateCreated;
		this.lastUpdate = lastUpdate;
	}

    public Product(String sku, String name, String description, double unitPrice, String imageUrl, boolean active, int unitInStock, Date dateCreated, Date lastUpdate) {
        this.sku = sku;
        this.name = name;
        this.description = description;
        this.unitPrice = unitPrice;
        this.imageUrl = imageUrl;
        this.active = active;
        this.unitInStock = unitInStock;
        this.dateCreated = dateCreated;
        this.lastUpdate = lastUpdate;
    }

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getSku() {
		return sku;
	}

	public void setSku(String sku) {
		this.sku = sku;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public double getUnitPrice() {
		return unitPrice;
	}

	public void setUnitPrice(double unitPrice) {
		this.unitPrice = unitPrice;
	}

	public String getImageUrl() {
		return imageUrl;
	}

	public void setImageUrl(String imageUrl) {
		this.imageUrl = imageUrl;
	}

	public boolean isActive() {
		return active;
	}

	public void setActive(boolean active) {
		this.active = active;
	}

	public int getUnitInStock() {
		return unitInStock;
	}

	public void setUnitInStock(int unitInStock) {
		this.unitInStock = unitInStock;
	}

	public Date getDateCreated() {
		return dateCreated;
	}

	public void setDateCreated(Date dateCreated) {
		this.dateCreated = dateCreated;
	}

	public Date getLastUpdate() {
		return lastUpdate;
	}

	public void setLastUpdate(Date lastUpdate) {
		this.lastUpdate = lastUpdate;
	}

	public ProductCategory getCategory() {
		return category;
	}

	public void setCategory(ProductCategory category) {

		System.out.println("set Category  "+category);
		ProductCategory tempCategory=new ProductCategory();
		tempCategory.setId(category.getId());
		tempCategory.setCategoryName(category.getCategoryName());
		this.category = tempCategory;
		System.out.println("this . set Category  "+this.category);
	}
}

File: 01-frontend/angular-ecommerce/src/app/common/product.ts
export class Product {
	private _id: string;
	public get id(): string {
		return this._id;
	}
	public set id(value: string) {
		this._id = value;
	}
	private _sku: string;
	public get sku(): string {
		return this._sku;
	}
	public set sku(value: string) {
		this._sku = value;
	}
	private _name: string;
	public get name(): string {
		return this._name;
	}
	public set name(value: string) {
		this._name = value;
	}
	private _description: string;
	public get description(): string {
		return this._description;
	}
	public set description(value: string) {
		this._description = value;
	}
	private _unitPrice: number;
	public get unitPrice(): number {
		return this._unitPrice;
	}
	public set unitPrice(value: number) {
		this._unitPrice = value;
	}
	private _imageUrl: string;
	public get imageUrl(): string {
		return this._imageUrl;
	}
	public set imageUrl(value: string) {
		this._imageUrl = value;
	}
	private _active: boolean;
	public get active(): boolean {
		return this._active;
	}
	public set active(value: boolean) {
		this._active = value;
	}
	private _unitInStock: number;
	public get unitInStock(): number {
		return this._unitInStock;
	}
	public set unitInStock(value: number) {
		this._unitInStock = value;
	}
	private _dateCreated: Date;
	public get dateCreated(): Date {
		return this._dateCreated;
	}
	public set dateCreated(value: Date) {
		this._dateCreated = value;
	}
	private _lastUpdate: Date;
	public get lastUpdate(): Date {
		return this._lastUpdate;
	}
	public set lastUpdate(value: Date) {
		this._lastUpdate = value;
	}
}
